File: 0231-power-of-two/0231-power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        '''if (n == 0):
            return False
        while (n != 1):
            if (n % 2 != 0):
                return False
            n = n // 2
             
        return True'''
        if(n<1):
            return False
    
        if(n==1):
            return True
        
        if(n%2==1):
            return False
        
        return self.isPowerOfTwo(n/2)
        

File: 0231-power-of-two/0231-power-of-two.java
class Solution {
    public boolean isPowerOfTwo(int n) {
         if (n <= 0) {
            return false;
        }
        return (n & (n - 1)) == 0;
    }
        //if(n<1){
            //return false;
        //}
        //if(n==1){
            //return true;
        //}
        //if(n%2==1){
            //return false;
        //}
        //return(isPowerOfTwo(n/2));
        
        
    }
